crates/bevy_ecs/src/bundle.rs

```diff
@@ -1295,6 +1318,7 @@ mod tests {
         world.init_resource::<R>();
         let mut entity = world.entity_mut(entity);
         entity.insert(A);
+        entity.insert_if_new(A); // this will not trigger on_replace or on_insert
```

Comment:
This comment is wrong, `on_insert` will still be called, and in fact the test fails.

Reply:
The comment (and test) is correct now: `on_insert` is only called for added or replaced components.

(I don't have a strong opinion on whether that's the desirable behavior, though. You could argue that `insert` was still called even if it didn't do anything, so the hook should be called too.)